ingestion/src/metadata/ingestion/source/database/oracle/metadata.py

```diff
@@ -186,30 +187,41 @@ def process_result(self, data: FetchProcedureList):
         result_dict = {}
 
         for row in data:
-            owner, name, line, text = row
+
+            owner, name, line, text, procedure_type = row
             key = (owner, name)
             if key not in result_dict:
-                result_dict[key] = {"lines": [], "text": ""}
+                result_dict[key] = {"lines": [], "text": "", "procedure_type": ""}
             result_dict[key]["lines"].append(line)
             result_dict[key]["text"] += text
+            result_dict[key]["procedure_type"] = procedure_type
 
         # Return the concatenated text for each procedure name, ordered by line
         return result_dict
 
+    def _get_stored_procedures_internal(
+        self, query: str
+    ) -> Iterable[OracleStoredProcedure]:
```

Comment:
OracleStoredProcedure doesn't seem right for the method which takes care of stored procedure and packages, let's make it generic for both